controllers/rhmi/rhmi_controller.go

```diff
@@ -231,6 +232,46 @@ func (r *RHMIReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error) {
 		installationCfgMap = installation.Spec.NamespacePrefix + DefaultInstallationConfigMapName
 	}
 
+	skuSecretName := ""
+	installationSKU := &sku.SKU{}
+	if installation.Spec.Type == string(rhmiv1alpha1.InstallationTypeManagedApi) {
+		skuSecretName, found, err := addon.GetStringParameterByInstallType(context.TODO(), r.Client, rhmiv1alpha1.InstallationTypeManagedApi, request.NamespacedName.Namespace, "sku")
+		if err != nil {
+			return retryRequeue, errors.Wrap(err, "Error checking for SKU secret")
+		}
+
+		if !found && !installation.ObjectMeta.CreationTimestamp.Time.Before(time.Now().Add(-(1 * time.Minute))) {
+			return retryRequeue, nil
+		}
+		//!found means the param wasn't found so we want to default rather than return
+		//but don't do it until the installation object is more than a minute old in case the secret is slow to create
+		if !found && installation.ObjectMeta.CreationTimestamp.Time.Before(time.Now().Add(-(1 * time.Minute))) {
+			skuSecretName = "TWENTY_MILLION_SKU"
+		}
+		// get a configmap from the cluster
+		cm := &corev1.ConfigMap{}
+		err = r.Get(context.TODO(), types.NamespacedName{Namespace: request.Namespace, Name: "sku-config"}, cm)
+		if err != nil {
+			return retryRequeue, errors.Wrap(err, "Error getting sku config map")
+		}
+		err = sku.GetSKU(skuSecretName, cm, installationSKU)
+		if err != nil {
+			return retryRequeue, err
+		}
+
+		// if it's the first install and first reconcile there will be no value
+		// set it because we want to be able to compare the sku
+		if installation.Status.ToSKU == "" && installation.Status.SKU == "" {
+			installationSKU.SetUpdated(true)
```

Comment:
If we do this in the `sku.GetSKU` it can be kept private so the SKU remains immutable.